app/services/report_importer.rb

```diff
@@ -100,9 +100,12 @@ def notify_on_report_error(mail_error_state)
       end
 
       owners = host.owner.present? ? host.owner.recipients_for(:puppet_error_state) : []
+      users = User.puppet_error_notifications
```

Comment:
This is trying to get all users with puppet error notifications that are authorized to view hosts. For the first part, `MailNotification::PuppetError.all.map(&
:users)` should work. 

You could extend the object to put a validation so that `view_hosts` is a required permission on users that have that mail notification as well, does that sound right? I'm not sure this permission check is very transparent though, this would need to be documented well or (better) we should warn on Foreman directly.